src/amcl/include/amcl/map/map.h

```diff
@@ -0,0 +1,150 @@
+/*
```

Comment:
We should consider the scope of the c++ classes and put the code in a corresponding location. For example, is this map class specific to amcl? Specific to localization? Generally useful to other modules as well? Depending on the answer, we would locate this class in a different location.

Reply:
Agreed.  I'm planning to work on this on phase 2.